mne_bids/mne_bids.py

```diff
@@ -551,26 +528,23 @@ def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
     of the participant correctly.
 
     """
-    if isinstance(raw_file, string_types):
-        # We must read in the raw data
-        raw = _read_raw(raw_file, electrode=electrode, hsp=hsp, hpi=hpi,
-                        config=config, verbose=verbose)
-        _, ext = _parse_ext(raw_file, verbose=verbose)
-        raw_fname = raw_file
-    elif isinstance(raw_file, BaseRaw):
-        # We got a raw mne object, get back the filename if possible
-        # Assume that if no filename attr exists, it's a fif file.
-        raw = raw_file.copy()
-        if hasattr(raw, 'filenames'):
-            _, ext = _parse_ext(raw.filenames[0], verbose=verbose)
-            raw_fname = raw.filenames[0]
-        else:
-            # FIXME: How to get the filename if no filenames attribute?
-            raw_fname = 'unknown_file_name'
-            ext = '.fif'
-    else:
-        raise ValueError('raw_file must be an instance of str or BaseRaw, '
-                         'got %s' % type(raw_file))
+    if not isinstance(raw, BaseRaw):
+        raise ValueError('raw_file must be an instance of BaseRaw, '
+                         'got %s' % type(raw))
+
+    # We got a raw mne object, get back the filename if possible
+    # Assume that if no filename attr exists, it's a fif file.
+    if not hasattr(raw, 'filenames'):
+        raise ValueError('raw.filenames is missing.')
+
+    raw_fname = raw.filenames[0]
+    if '.ds' in op.dirname(raw.filenames[0]):
+        raw_fname = op.dirname(raw.filenames[0])
+    _, ext = _parse_ext(raw_fname, verbose=verbose)
```

Comment:
@sappelhoff I need your help here. For `.vhdr` files, `raw.filenames[0]` has the extension `.eeg` and not `.vhdr`. Is this expected?

Reply:
yes, this is how MNE-Python handles the `filenames` attribute. It places the datafile there, which in the case of BrainVision is the `.eeg`.